Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -0,0 +1,161 @@
+#ifndef WARPX_PEC_KERNELS_H_
+#define WARPX_PEC_KERNELS_H_
+
+#include "WarpX.H"
+#include "Utils/WarpXAlgorithmSelection.H"
+
+
+namespace PEC {
+using namespace amrex;
+    /**
+     * \brief Determines if the field boundary condition stored in fboundary is PEC
+     *        in direction, dir, is PEC.
+     *
+     * \param[in] fboundary  Value containing boundary type
+     * \param[in] dir        direction
+     *
+     * \param[out] 1 if the boundary type is PEC else 0
+     */
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    int is_boundary_PEC (amrex::GpuArray<int, 3> const& fboundary, int dir) {
+        return int( fboundary[dir] == FieldBoundaryType::PEC);
+    }
+
+    /**
+     * \brief Sets the electric field value tangential to the PEC boundary to zero. The
+     *        field value of the guard cells outside the domain boundary are set equal
+     *        and opposite to the field in the valid cells at their mirrored locations.
+     *        The number or depth of guard cells updated is equal to the shape factor of
+     *        particles in each dimension.
+     *
+     * \param[in] icomp        component of the Efield being updated
+                               (0=x, 1=y, 2=z in 3D) and (0=x, 1=z in 2D)
+     * \param[in] dom_lo       index value of the lower domain boundary (cell-centered)
+     * \param[in] dom_hi       index value of the higher domain boundary (cell-centered)
+     * \param[in] ijk_vec      indices along the x(i), y(j), z(k) of Efield Array4
+     * \param[in] Efield       field data to be updated if (ijk) is at the boundary or a guard cell
+     * \param[in] stag         staggering of the field data being updated.
+     * \param[in] fbndry_lo    Field boundary type at the lower boundaries
+     * \param[in] fbndry_hi    Field boundary type at the upper boundaries
+     */
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    void SetTangentialEfield (const int icomp, const amrex::IntVect & dom_lo,
+                                const amrex::IntVect &dom_hi,
+                                const amrex::IntVect &ijk_vec,
+                                amrex::Array4<amrex::Real> const& Efield,
+                                const amrex::IntVect& stag,
+                                amrex::GpuArray<int, 3> const& fbndry_lo,
+                                amrex::GpuArray<int, 3> const& fbndry_hi )
+    {
+        // Fields in guard cells set equal and opposite to valid cells
+        amrex::IntVect ijk_valid = ijk_vec;
+        for (int idim = 1; idim < AMREX_SPACEDIM; ++idim) {
+            // index of dimension
```

Comment:
```suggestion
            // index of dimension
            // the loop over idim causes dir to take the value of each of the tangential directions
```

Reply:
this part changed and I added more comments here to clarify the changes and what dir is